packages/core/src/components/steps-modal/steps-modal.tsx

```diff
@@ -105,7 +122,25 @@ export class AtomStepsModal {
     e.stopImmediatePropagation()
     this.isOpen = false
     this.atomDidDismiss.emit(this.currentStep)
-    this.currentStep = 1
+    this.currentStep = this.customInitialStep ? this.customInitialStep : 1
+  }
+
+  private get secondaryButtonText() {
+    return this.currentStep === this.customInitialStep
```

Comment:
Shouldn't we remove the `cancelButtonText` as there is the `secondaryButtonTextsArray`?

Reply:
the cancel button text is a facilitator to allow user to set the secondary button text directly when it is on first step, so the user doesn't need to set a new array of strings for each time the customInitialStep is set, they only set the default cases and this cancel, so they can change the custom initial step to wherever he wants without need a new array

Reply:
Sorry, man. I don't know if I got your point. Once the array is given, isn't always passed anyway?